### ISSUE TITLE
Change sensor data format to save on disk space

### ISSUE THREAD

**Issue by reyery** [architecture-building-systems/CityEnergyAnalyst#3304]
Sensor data from radiation script is now being saved in JSON format. When running the script on a larger scale, these files add up and might consume a lot of disk space. This could be solved by saving the files in another format that supports compression. Since users rarely open sensor data file manually, I think it is a good compromise to save it in another format while removing the readability of the file (i.e. opening it in a text editor). 

I just did a simple size comparison for 1012 buildings, using different formats for the sensor data with compression (hdf5 and feather). 
We could achieve about 4-6x reduction in size with other formats:
```
feather    6.379GB
h5         4.333GB
json      24.212GB
```

I would suggest using feather since it is consumes much lesser CPU resources (while reading and writing) compared to hdf5 even though the size is not as small.

**Comment by jimenofonseca** [comment]
beautiful idea. I think we do not need to make these files human readable and go straight away to h5. Does pandas read it as easy as a json or .csv? that will be important.

**Comment by reyery** [comment]
Yes, in fact pandas official supports reading/writing to both hdf5 and feather format

**Comment by jimenofonseca** [comment]
What about writing, reading time to and from dataframe?I remember trying frather and H5 for solar radiation about 5 years ago and stopped due to this.Was this improved?Sent from my iPhoneOn 6 May 2023, at 16:19, Reynold Mok ***@***.***> wrote:﻿
Yes, in fact pandas official supports reading/writing to both hdf5 and feather format

**Comment by reyery** [comment]
It's actually faster to read/write feather compared to json. The time taken for feather is quite consistent and is sub 0s even for large buildings.

For a single building with 2917 sensors
```
write json       20.513s
read json         4.054s

write feather     0.582s
read feather      0.182s

size json        305.4MB
size feather      40.3MB
```

For a single building with 217 sensors
```
write json       1.4673s
read json         0.265s

write feather     0.507s
read feather      0.126s

size json         22.7MB
size feather       6.6MB
```

**Comment by jimenofonseca** [comment]
Then that is it. Lets do it. One can also think of defining a right dtype for each argument of the data. It saves dramatically space. I saw some work in a data science competition which automates that.On 6 May 2023, at 17:49, Reynold Mok ***@***.***> wrote:﻿
It's actually faster to read/write feather compared to json. The time taken for feather is quite consistent and is sub 0s even for large buildings.
For a single building with 2917 sensors
write json       20.513s
read json         4.054s

write feather     0.582s
read feather      0.182s

size json        305.4MB
size feather      40.3MB

For a single building with 217 sensors
write json       1.4673s
read json         0.265s

write feather     0.507s
read feather      0.126s

size json         22.7MB
size feather       6.6MB

**Comment by reyery** [comment]
I have already implemented that for the sensor data before and it helps with memory usage when loading the data :thumbsup:
